mono/metadata/metadata.c

```diff
@@ -3893,14 +3893,14 @@ mono_type_size (MonoType *t, int *align)
 		return 4;
 	case MONO_TYPE_I8:
 	case MONO_TYPE_U8:
-#if defined(PLATFORM_IPHONE_XCOMP)
+#if defined(PLATFORM_IPHONE) || defined(PLATFORM_IPHONE_XCOMP)
```

Comment:
This change scares me a bit. The idea is that the cross compiler needs to force the same alignment as the native runtime has by default. That means I would not expect the check for `PLATFORM_IPHONE` is needed, since that is the case of the native ARM runtime.

Reply:
It looks like the native runtime itself has to consult this function sometimes, so not having PLATFORM_IPHONE results in returning wrong data on the runtime side.

Reply:
It's fine it consults it. The whole assumption here is that `__alignof__(gint64)` is always 4 on the iOS runtime.

Reply:
That is what the `#if` is forcing in the cross compiler. This PR seems to indicate that is not true.

Reply:
Ah, understood. Indeed, now it looks scary to me too. Did a quick check and it seems that __alignof__ is broken for 64-bit types. This code:

```	std::cout << __alignof__(int64_t) << "\n";
	
	struct Test {
		int x;
		int64_t y;
	};
	std::cout << sizeof(Test) << "\n";``` returns 8 and 12 on iOS armv7.

Reply:
What do you mean `returns 8 and 12 on iOS armv7.`? Which does it return?

Reply:
`__alignof__(int64_t) == 8`, but `sizeof(Test) == 12`. This means that `__alignof__` returns wrong data, because if the alignment of `int64_t` was 8 bytes, then `sizeof(Test)` would have been 16. I vaguely remember fixing similar issue in another place, let me dig it up.

Reply:
Here it is: 2f064e89d916d82765e84a485b610807d7e87af8.

Reply:
Do we want to fix this issue for PLATFORM_IPHONE in the same way?

Reply:
Yes, I think that makes sense.